source/slang/slang-ir-legalize-varying-params.h

```diff
@@ -21,13 +22,63 @@ void legalizeEntryPointVaryingParamsForCUDA(
     IRModule*               module,
     DiagnosticSink*         sink);
 
-IRInst* emitCalcGroupThreadIndex(
-    IRBuilder& builder,
-    IRInst* groupThreadID,
-    IRInst* groupExtents);
+
+// TODO: Once `slang-ir-metal-legalize.cpp` is merged with 
+// `slang-ir-legalize-varying-params.cpp`, move the following
+// below into `slang-ir-legalize-varying-params.cpp` as well
 
 IRInst* emitCalcGroupExtents(
     IRBuilder& builder,
     IRFunc* entryPoint,
     IRVectorType* type);
+
+IRInst* emitCalcGroupIndex(
+    IRBuilder& builder,
+    IRInst* groupThreadID,
+    IRInst* groupExtents);
+
+// SystemValueSemanticName member definition macro
+#define SYSTEM_VALUE_SEMANTIC_NAMES(M)              \
```

Comment:
It looks like this macro is used only in `source/slang/slang-ir-legalize-varying-params.cpp`.
Maybe you want to move this to the cpp file.
I see that you left a similar comment a few lines above but it is unclear which lines "the following below" refers to.

Reply:
> It looks like this macro is used only in source/slang/slang-ir-legalize-varying-params.cpp.
> Maybe you want to move this to the cpp file.

This will not work. `SYSTEM_VALUE_SEMANTIC_NAMES` is a helper macro which is used to help define members of `SystemValueSemanticName`. `SystemValueSemanticName` members are used by `slang-ir-metal-legalize.cpp` and `slang-ir-legalize-varying-params.cpp`.

> I see that you left a similar comment a few lines above but it is unclear which lines "the following below" refers to.

This "similar" comment refers to functionality requested by #4375. I will adjust the comment accordingly to be clearer.